primitives/src/treasury_manager.rs

```diff
@@ -0,0 +1,40 @@
+// Ajuna Node
+// Copyright (C) 2022 BlogaTech AG
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+use frame_support::{
+	pallet_prelude::{DispatchError, Member},
+	Parameter,
+};
+use parity_scale_codec::Codec;
+
+/// The treasury manager trait that can be passed around to other pallets that need to works with
+/// the treasury or its accounts
+pub trait TreasuryManager {
+	type AccountId: Member + Codec;
+
+	type Currency: Member + Codec;
+
+	type TreasuryKey: Parameter + Member;
```

Comment:
It's not exactly an account, the key is just a way to identify a given treasury pot, since we do have some cases in which we split the treasury based on criteria like the season_id for example

Reply:
Hmm maybe more lie `TreasuryPotKey`?